core/iwasm/interpreter/wasm_runtime.c

```diff
@@ -1422,8 +1431,8 @@ execute_malloc_function(WASMModuleInstance *module_inst, WASMExecEnv *exec_env,
         /* the malloc functino from assemblyscript is:
             function __new(size: usize, id: u32)
             id = 0 means this is an ArrayBuffer object */
-        argv[1] = 0;
-        argc = 2;
+        argv[argc] = 0;
+        argc++;
```

Comment:
In loader, the __new function's arguments for memory64 are treated as int64 type, here we should also add extra control for memory64?

Reply:
And if that, should defined argv as argv[4]

Reply:
I am not sure if the assemblyscript supports memory64 yet, according to https://github.com/AssemblyScript/assemblyscript/wiki/Status-and-Roadmap it's not supported yet, so maybe consider throwing a warning or error here when memory64 is enabled.

Reply:
> In loader, the __new function's arguments for memory64 are treated as int64 type, here we should also add extra control for memory64?

I don't think we need to do any change. As far as I understand, we'll have one of those two scenarios:
* wasi libc - in that case, only `malloc(uint64) -> uint64` is called
* assemblyscript - for this, two functions are called: 1) `__new(uint64, uint32) -> uint64` and then `__pin(uint64) -> uint64`

Since we re-use the same array for __new and __pin (and __new leaves the last return value as first argument(s) in the argv so it can be used for __pin) I don't think we'll need to make any further changes.

> I am not sure if the assemblyscript supports memory64 yet, according to Wiki: Status and Roadmap (AssemblyScript/assemblyscript) it's not supported yet, so maybe consider throwing a warning or error here when memory64 is enabled.

Hmm, I'm personally a bit against it. If we do that, we'll need to keep track on assemblyscript roadmap, and it'd be easy to miss. At the same time, if memory64 is not supported yet by AS, users will unlikely ever have AS binary compiled to wasm64. If they do, they likely did some ugly hacks and they're on their own anyway.

Reply:
Treating __new and __pin in asc as `1) __new(uint64, uint32) -> uint64 and then __pin(uint64) -> uint64` is good, but I think the type of the second argument of __new in loader/mini_loader is invalid: it uses VALUE_TYPE_I64 now.

Reply:
you're right, updated.